storage-checks.sh

```diff
@@ -0,0 +1,349 @@
+#!/bin/bash
+
+: '
+This scipt will validate blob storage from AIC perspective.
+# $1 - json file with credentials
+[Script Version -> 21.4]
+Credentials file structure
+{"AWS_HOST": <AWS_HOST>, "AWS_ENDPOINT": <AWS_ENDPOINT>, "AWS_ACCESS_KEY_ID": <AWS_ACCESS_KEY_ID>, "AWS_SECRET_ACCESS_KEY": <AWS_SECRET_ACCESS_KEY>, "BUCKET_1": <BUCKET_1>, "BUCKET_2": <BUCKET_2>}
+where the access_key and secret correspond to aws crednetials which has access to the two buckets (customer creates buckets with appropriate policies) 
+Requirements: aws s3, jq
+'
+
+red=$(tput setaf 1)
+green=$(tput setaf 2)
+yellow=$(tput setaf 3)
+default=$(tput sgr0)
+
+declare -A etags
+
+echo "$green $(date) Starting validation of object storage $default"
+
+readonly CREDENTIALS_FILE=$1
+
+function echoinfo() {
+	echo "$yellow $(date) $1 $default"
+}
+
+function errecho() {
+  echo "$red $(date) $1 $default"
+}
+
+# Validate file provided by user exists or not, It may be relative path or absolute path
+# $1 - File path
+function validate_file_path() {
+  if [ ! -f "$1" ]; then
+    echo "$red $(date) $1 file does not exist, Please check ... Exiting $default"
+    exit 1
+  fi
+}
+
+# Validate dependency module
+# $1 - Name of the dependency module
+# $2 - Command to validate module
+function validate_dependency() {
+  eval $2
+  # Next statement is checking last command success aws --version has some issue
+  if [ $? -ne 0 ]; then
+    echo "$red $(date) Please install ******** $1 ***********  ... Exiting $default"
+    exit 1
+  fi
+}
+
+# Validate required modules exits in target setup
+function validate_setup() {
+  validate_dependency "aws s3" "aws --version"
+  validate_dependency "jq" "jq --version"
+  #validate_dependency "fallocate" "fallocate -V"
+  validate_dependency "yes" "yes --version"
+  echo "$(date) Successfully validated required dependencies"
+}
+
+function initialize_variables() {
+  # Validate file path
+  validate_file_path $CREDENTIALS_FILE
+
+  export AWS_HOST=$(cat $CREDENTIALS_FILE | jq -r 'select(.AWS_HOST != null) | .AWS_HOST')
```

Comment:
can we check if credential file is valid json and return appropriate message before this step